drafts/2019-10-22-liberty-dev-mode.adoc

```diff
@@ -0,0 +1,66 @@
+---
+= Open Liberty development mode
+Eric Lau <https://github.com/ericglau>
+
+Open Liberty development mode, or dev mode, allows you to develop your application with any text editor or IDE.  Your code will be automatically compiled and deployed to your running server, making it easy to iterate on your changes. You will be able to run tests on demand or even automatically so that you can get immediate feedback on your changes. You will also be able to attach a debugger at any time to debug your running application.
+
+For a quick demo of what you can do with dev mode, check out the following video:
+
+++++
+<iframe width="560" height="315" src="https://www.youtube.com/embed/eetnJrhVOMM" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+++++
+
+You can also follow the steps in the https://github.com/OpenLiberty/demo-devmode[demo-devmode] project used in the above video to try it out yourself.
+
+== Project setup
+
+For Maven projects, dev mode is run through the Liberty Maven Plugin.  Specify the latest version of the Liberty Maven Plugin in your project's pom.xml file.  Note that in versions 3.0 and above, the groupId of the plugin is `io.openliberty.tools`.
+[source,xml]
+----
+    <plugin>
+        <groupId>io.openliberty.tools</groupId>
+        <artifactId>liberty-maven-plugin</artifactId>
+        <version>3.1</version>
+    </plugin>
+----
+
+== Starting dev mode
+
+Start dev mode by running the following command in your project:
+----
+mvn liberty:dev
+----
+
+== Hot deployment
+
+Once dev mode is started, source code changes will be automatically compiled and deployed to the running server.  You can add new dependencies in your pom.xml file to be used for compilation.  Resources and server configuration changes will also be picked up.  If any features are added to the server.xml file, they will be installed automatically.  
+
+== On demand testing and hot testing
+
+To run tests on demand, press Enter in the terminal where dev mode is running.  This will invoke the unit tests and integration tests which are configured for your project.
+
+To run tests automatically after every change, include the `-DhotTests` parameter when starting dev mode.  For example:
+----
+mvn liberty:dev -DhotTests
+----
+
+== Debugger support
+
+At any point during dev mode, you can attach a Java debugger to debug your application.  By default, the debug port is `7777`, but another port will be used and shown in the console output if the default port is not available.
+
+== Stopping dev mode
+
+You can quit dev mode at any time by pressing `Control-C` in the terminal, or type `q` and press Enter.
+
+== Further reading
+
+For more details on how to use dev mode, see the documentation for the https://github.com/OpenLiberty/ci.maven/blob/master/docs/dev.md#dev[dev] goal of the Liberty Maven Plugin.
```

Comment:
Make the link longer and more informative: "...[dev goal of the Liberty Maven Plugin]."